credentials/alts/internal/handshaker/service/service.go

```diff
@@ -58,3 +58,21 @@ func Dial(hsAddress string) (*grpc.ClientConn, error) {
 	}
 	return hsConn, nil
 }
+
+// Close closes all open connections to the handshaker service.
+//
+// For testing purposes only.
```

Comment:
This seems like a generic close function. Why is it expected to be used in tests only?

Reply:
In practice, the connection to the handshaker service is kept open until the application is killed, so this Close function is never needed/called.

However, in the e2e test, we need to do call this Close function so that we don't get a goroutine leak.

Reply:
Why can't you explicitly close the connection from the test?

Reply:
Disclaimer: I'm not a Go expert so apologies if I'm missing something obvious. :)

The [`hsConnMap`](https://github.com/grpc/grpc-go/blob/3489bb7d51741618f1b1e0c723ca6dc905118e3b/credentials/alts/internal/handshaker/service/service.go#L36), which holds the connection, is not exported. So we have 2 options:

1. Add a test-only `Close` API to close any connections in the map, which we call in `TestFullHandshake`.
2. Call `Dial` to get the same connection and then manually close it. This feels hacky, because we never want users to call `Dial` (it should only be called by the internals of the ALTS libraries) and so I didn't want to set this precedent).

WDYT?

Reply:
Thanks for the exaplaination. How about calling the method `CloseForTesting` or something which makes is very obvious that the method is only for testing purposes. We do that regularly in our codebase. Thanks.

Reply:
Sure thing, done.